modules/eslint-plugin/spec/rules/component-store/require-super-ondestroy.spec.ts

```diff
@@ -0,0 +1,110 @@
+import { ESLintUtils } from '@typescript-eslint/utils';
+import { InvalidTestCase, ValidTestCase } from '@typescript-eslint/rule-tester';
+import rule, {
+  messageId,
+} from '../../../src/rules/component-store/require-super-ondestroy';
+import { fromFixture, ruleTester } from '../../utils';
+import path = require('path');
+
+type MessageIds = ESLintUtils.InferMessageIdsTypeFromRule<typeof rule>;
+type Options = ESLintUtils.InferOptionsTypeFromRule<typeof rule>;
+
+const valid: () => (string | ValidTestCase<Options>)[] = () => [
```

Comment:
Thanks, @markoblagdan this is looking good!

> I do not handle the case when ComponentStore is a custom class not imported from @ngrx/component-store, which means this rule will raise a false error if one extends from a custom ComponentStore class and overrides ngOnDestroy. Seemed like a weird edge case so I did not handle it, but let me know if it would be better to do so anyway

Do you want to add a test for this case please, that way it's also "documented"

Reply:
Of course, added valid and invalid cases for that scenario, but I wonder if I should leave a comment at the test case(s), maybe linking to this comment or PR?

Reply:
Feel free to add a comment.
I had a different test in my mind though:

```ts
import { BookStore } from '../components/component-store';
class BooksStore extends BookStore implements OnDestroy
{
  cleanUp() {}
  override ngOnDestroy(): void {
    this.cleanUp();
    super.ngOnDestroy();
  }
}
```

Reply:
That will not raise a false error, that is, no error will be shown when _not_ calling super.ngOnDestroy in your example. The error will be wrongly raised only if the extended class is literally called `ComponentStore`.

So I'm not sure it makes sense to add your example, unless I'm missing something, since it doesn't (and shouldn't) trigger the rule.

Should I keep the case I've added?

Reply:
> Of course, added valid and invalid cases for that scenario, but I wonder if I should leave a comment at the test case(s), maybe linking to this comment or PR?

Was bothered by this so I handled the case with the rule after all, it wasn't so hard. Basically now I'm also checking if `ComponentStore` is imported from ngrx/component-store. Also updated thevalid test case for this.

Reply:
Nice, that's a good solution!